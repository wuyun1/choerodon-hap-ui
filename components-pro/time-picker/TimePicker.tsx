import DatePicker from '../date-picker/DatePicker';
import { ViewMode } from '../date-picker/enum';

export default class TimePicker extends DatePicker {
  static displayName = 'TimePicker';

  static defaultProps = {
    ...DatePicker.defaultProps,
    mode: ViewMode.time,
  };
}
