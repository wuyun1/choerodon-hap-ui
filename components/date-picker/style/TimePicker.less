.@{calendar-timepicker-prefix-cls} {
  position: absolute;
  width: 100%;
  top: .4rem;
  background-color: @component-background;

  &-panel {
    z-index: @zindex-picker;
    position: absolute;
    width: 100%;
  }

  &-inner {
    display: inline-block;
    position: relative;
    outline: none;
    list-style: none;
    font-size: @font-size-base;
    text-align: left;
    background-color: @component-background;
    background-clip: padding-box;
    line-height: 1.5;
    overflow: hidden;
    width: 100%;
  }
  &-combobox {
    width: 100%;
  }

  &-column-1,
  &-column-1 &-select {
    width: 100%;
  }
  &-column-2 &-select {
    width: 50%;
  }
  &-column-3 &-select {
    width: 33.33%;
  }
  &-column-4 &-select {
    width: 25%;
  }

  &-input-wrap {
    display: none;
  }

  &-select {
    float: left;
    font-size: @font-size-base;
    border-right: @border-width-base @border-style-base @border-color-split;
    box-sizing: border-box;
    overflow: hidden;
    position: relative; // Fix chrome weird render bug
    height: 2.26rem;

    &:hover {
      overflow-y: auto;
    }

    &:first-child {
      border-left: 0;
      margin-left: 0;
    }

    &:last-child {
      border-right: 0;
    }

    ul {
      list-style: none;
      box-sizing: border-box;
      margin: 0;
      padding: 0;
      width: 100%;
      max-height: 2.06rem;
    }

    li {
      text-align: center;
      list-style: none;
      box-sizing: content-box;
      margin: 0;
      width: 100%;
      height: .24rem;
      line-height: .24rem;
      cursor: pointer;
      user-select: none;
      transition: background 0.3s ease;
    }

    li:last-child:after {
      content: '';
      height: 2.02rem;
      display: block;
    }

    li:hover {
      background: @item-hover-bg;
    }

    li&-option-selected {
      background: @time-picker-selected-bg;
      font-weight: bold;
    }

    li&-option-disabled {
      color: @btn-disable-color;
      &:hover {
        background: transparent;
        cursor: not-allowed;
      }
    }
  }
}

.@{calendar-prefix-cls}-time {
  .@{calendar-prefix-cls}-day-select {
    padding: 0 .02rem;
    font-weight: 500;
    display: inline-block;
    color: @heading-color;
    line-height: .34rem;
  }

  .@{calendar-prefix-cls}-footer {
    position: relative;
    height: auto;

    &-btn {
      text-align: right;
    }

    .@{calendar-prefix-cls}-today-btn {
      float: left;
      margin: 0;
    }

    .@{calendar-prefix-cls}-time-picker-btn {
      display: inline-block;
      margin-right: .08rem;

      &-disabled {
        color: @disabled-color;
      }
    }
  }
}
