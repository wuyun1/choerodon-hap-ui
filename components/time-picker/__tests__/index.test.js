import React from 'react';
import { mount, render } from 'enzyme';
import RcTimePicker from '../../rc-components/time-picker/TimePicker';
import TimePicker from '..';
import focusTest from '../../../tests/shared/focusTest';

describe('TimePicker', () => {
  focusTest(TimePicker);

  it('renders addon correctly', () => {
    const addon = () => (<button type="button">Ok</button>);
    const wrapper = mount(<TimePicker addon={addon} />);
    const rcTimePicker = wrapper.find(RcTimePicker);
    const addonWrapper = render(rcTimePicker.props().addon());

    expect(addonWrapper).toMatchSnapshot();
  });
});
