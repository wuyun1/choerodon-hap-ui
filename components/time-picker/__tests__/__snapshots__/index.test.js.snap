// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker renders addon correctly 1`] = `
<div
  class="c7n-time-picker-panel-addon"
>
  <button
    type="button"
  >
    Ok
  </button>
</div>
`;
